ip_country/src/bit_queue.rs

```diff
@@ -0,0 +1,413 @@
+use std::collections::VecDeque;
+
+#[derive(Debug)]
+pub struct BitQueue {
+    back_blank_bit_count: usize, // number of high-order bits in the back byte of the queue that are unused
+    byte_queue: VecDeque<u8>,
+    front_blank_bit_count: usize, // number of low-order bits in the front byte of the queue that are unused
+}
+
+impl BitQueue {
+    pub fn new() -> Self {
+        let byte_queue = VecDeque::from(vec![0, 0]);
+        Self {
+            back_blank_bit_count: 8,
+            byte_queue,
+            front_blank_bit_count: 8,
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        (self.byte_queue.len() * 8) - self.back_blank_bit_count - self.front_blank_bit_count
+    }
+
+    #[allow(unused_assignments)]
```

Comment:
Honestly, I don't know why this is a thing. Can you confirm the lint treats an issue which is still somewhere here?

Regarding how you can proceed, you can try compiling this crate by its `ci/all.sh` or maybe `ci/unit_tests.sh`, ideally when you've completed the job on the other comments and now you have quite clean code. Comment out the lint and try compiling again. If it throws out a warning (and probably kills the build) then it may still be necessary. On the other hand, it could also hint you how to resolve this by enhancement of the code itself, if you're lucky to meet Clippy and its wittiness. 

If you've embarked on this then maybe scroll right away down to line 46 and test it on as well as you did here.